apps/web/src/features/accounts/components/account-sections/account-activity.tsx

```diff
@@ -54,7 +54,14 @@ type TransactionData = {
 
 const getTransaction = ({ network, txHash }: { network: string; txHash: string }) => {
   return async (): Promise<TransactionData> => {
-    const provider = new JsonRpcProvider(network);
+    const provider = new JsonRpcProvider(network || 'https://eth.llamarpc.com');
```

Comment:
can we create a unique method that creates a provider. This code is copied on multiple places

Reply:
it should be:

```
async function createProvider(){
const prov =new JsonRpcProvider('https://eth.llamarpc.com'); 
    try {
      await provider._detectNetwork();
    } catch (err) {
      console.log('-----personalSign-----');
      console.log(err);
      provider.destroy();
    }
}

const provider = createProvider()
```